File: two_pointers/TwoPointers.kt
fun twoSum(arr: IntArray, target: Int): IntArray {
    var left = 0
    var right = arr.size - 1
    while (left < right) {
        val sum = arr[left] + arr[right]
        when {
            sum == target -> return intArrayOf(left, right)
            sum < target -> left++
            else -> right--
        }
    }
    return intArrayOf()
}

fun main() {
    val result = twoSum(intArrayOf(1, 2, 3, 4, 6), 6)
    println("[${result[0]}, ${result[1]}]")
}

File: two_pointers/two_pointers.js
function twoSum(arr, target) {
    let left = 0, right = arr.length - 1;
    while (left < right) {
        const sum = arr[left] + arr[right];
        if (sum === target) return [left, right];
        sum < target ? left++ : right--;
    }
    return [];
}

console.log(twoSum([1, 2, 3, 4, 6], 6)); // [1, 3]
